nautobot_floor_plan/template_content.py

```diff
@@ -43,18 +43,41 @@ def buttons(self):
 </a>"""
 
     def detail_tabs(self):
-        """Add a "Floor Plan" tab to the Location detail view if the Location has a FloorPlan associated to it."""
+        """Add a "Floor Plan" tab to the Location detail view or a "Child Floor Plan" tab if the Location has Children with FloorPlans associated to them."""
         location = self.context["object"]
-        try:
-            location.floor_plan  # pylint: disable=pointless-statement
-            return [
-                {
-                    "title": "Floor Plan",
-                    "url": reverse("plugins:nautobot_floor_plan:location_floor_plan_tab", kwargs={"pk": location.pk}),
-                }
-            ]
-        except ObjectDoesNotExist:
-            return []
+
+        # Tabs to return
+        tabs = []
+
+        # Check if the location has a floor plan
+        has_floor_plan = getattr(location, "floor_plan", None) is not None
+
+        # Check if the location has children locations with floor plans
+        has_child_floor_plans = location.children.filter(floor_plan__isnull=False).exists()
+        floor_plan_tab = {
+            "title": "Floor Plan",
+            "url": reverse(
+                "plugins:nautobot_floor_plan:location_floor_plan_tab",
+                kwargs={"pk": location.pk},
+            ),
+        }
+
+        child_tab = {
+            "title": "Child Floor Plan(s)",
+            "url": reverse("plugins:nautobot_floor_plan:location_child_floor_plan_tab", kwargs={"pk": location.pk}),
+        }
+        # If location has no children
+        if not location.children.exists():
+            if has_floor_plan:
+                tabs.append(floor_plan_tab)
+            return tabs
+        # If location has children with floor plans
+        if has_child_floor_plans:
+            if has_floor_plan:
+                tabs.append(floor_plan_tab)
+            tabs.append(child_tab)
+            return tabs
+        return tabs
```

Comment:
What if a location has a floor plan and has children, but no child floor plans? I think it would fail to add tab for the location floor plan with this logic, yes?

Does something like this work or is there a condition that it doesn't solve?
```suggestion
        if not has_floor_plan:
            return tabs

        tabs.append(floor_plan_tab)

        # If location has children with floor plans
        if has_child_floor_plans:
            tabs.append(child_tab)
        return tabs
```